### PR TITLE
[BNNS]Implement of resize_bilinear.

### PR DESCRIPTION
**Pull request by lisa0314**
Fix [intel/webml-polyfill#340](https://github.com/intel/webml-polyfill/issues/340).
@fujunwei , please help to review.

### CODE REVIEW

File: services/ml/compilation_impl_mac_bnns.mm [hunk 1]
```diff
@@ -11,6 +11,22 @@
 
 namespace ml {
 
+void ResamplingkernelFunc(const float* xArray,
+                          float* yArray,
+                          unsigned long count,
+                          void* userData) {
```
Comment (fujunwei) [inline services/ml/compilation_impl_mac_bnns.mm]: Unused parameter `userData`.

File: services/ml/execution_impl_mac_bnns.mm [hunk 2]
```diff
@@ -298,6 +298,90 @@ bool TransposeForInput(const ml::OperationMac& operation,
...
+              source_buffer.rowBytes = input_width * sizeof(float);
+
+              const OperandMac& output =
+                  compilation_->operands_[operation.outputs[0]];
+              const int32_t output_batch = output.dimensions[0];
+              const int32_t output_width = output.dimensions[1];
+              const int32_t output_height = output.dimensions[2];
+              const int32_t output_depth = output.dimensions[3];
+
+              float kernelWidth = 1;
```
Comment (fujunwei) [inline services/ml/execution_impl_mac_bnns.mm]: `kernelWidth  `=> `kernel_width`

File: services/ml/execution_impl_mac_bnns.mm [hunk 3]
```diff
@@ -298,6 +298,90 @@ bool TransposeForInput(const ml::OperationMac& operation,
...
+              float scale = float(output_height * output_depth * output_batch) /
+                            (input_height * input_depth * input_batch);
+
+              ResamplingFilter resamplingFilterForVerticle =
```
Comment (fujunwei) [inline services/ml/execution_impl_mac_bnns.mm]: Ditto, [camel case](https://chromium.googlesource.com/chromium/src/+/master/styleguide/c++/blink-c++.md#naming)

File: services/ml/execution_impl_mac_bnns.mm [hunk 4]
```diff
@@ -298,6 +298,90 @@ bool TransposeForInput(const ml::OperationMac& operation,
...
+                DLOG(ERROR) << "Fail to new vertical shear!";
+              }
+              vImage_Buffer des_buffer;
+              des_buffer.height = output_height;
+              des_buffer.width = output_width;
+              des_buffer.rowBytes = output_width * sizeof(float);
+
+              scale = (float)output_width / input_width;
+
+              ResamplingFilter resamplingFilterForHorizontal = malloc(
```
Comment (fujunwei) [inline services/ml/execution_impl_mac_bnns.mm]: It's better to extract resampling filter for vertical and horizontal to a function.

File: services/ml/compilation_impl_mac_bnns.mm (hunk 1 above)
Comment (lisa0314) [reply services/ml/compilation_impl_mac_bnns.mm]: The function parameter is defined in a function pointer. Although we don't use `useData` in our situation, we still need to include it.

File: services/ml/execution_impl_mac_bnns.mm (hunk 2 above)
Comment (lisa0314) [reply services/ml/execution_impl_mac_bnns.mm]: Yes

**Comment by fujunwei** [comment]
LGTM.

**Comment by huningxin** [comment]
Thanks for the PR and review!

**Pull request merged**